src/helpers/getDefaultBranch.js

```diff
@@ -1,10 +1,19 @@
 const runCommand = require("./runCommand");
 
+const BRANCHES = {
+	main: false,
+	master: false
+};
+
 const getDefaultBranch = () => {
-	const args = `remote show upstream | grep "HEAD branch" | sed 's/.*: //'`;
-	return runCommand({ args, showOutput: false }).then(response =>
-		response.trim()
-	);
+	const args = `branch -r | grep upstream | sed 's/^.*upstream\\///'`;
+	return runCommand({ args, showOutput: false }).then(response => {
+		const resultBranches = response.trim().split("\n");
+		Object.keys(BRANCHES).forEach(key => {
+			BRANCHES[key] = resultBranches.includes(key);
+		});
+		return Object.keys(BRANCHES).filter(key => BRANCHES[key])[0];
+	});
 };
```

Comment:
The code looks to works, but the logic can be simplified a little... see below. Also, we talked bout it blowing up with an error saying which branches are required if it cannot find one of the approved list.

```
const BRANCHES = [ "main", "master" ];

const getDefaultBranch = () => {
	const args = `branch -r | grep upstream | sed 's/^.*upstream\\///'`;
	return runCommand({ args, showOutput: false }).then(response => {
		const resultBranches = response.trim().split("\n");
		return BRANCHES.find( b => resultBranches.includes( b ) );
	});
};
```